tests/inc/Mocks/MockDataSource.php

```diff
@@ -30,4 +30,21 @@ public static function create( ?array $config = self::MOCK_CONFIG, ?ValidatorInt
 	public function set_endpoint( string $endpoint ): void {
 		$this->config['endpoint'] = $endpoint;
 	}
+
+	/**
+	 * Override the migrate_config method to adjust the config for testing.
+	 */
+	public static function migrate_config( array $config ): array|WP_Error {
```

Comment:
Added in this `migrate_config` implementation that helps to setup a successful and unsuccessful migration.

The idea is that all the other uses of this mock data source won't be impacted as they won't have this test field set or won't care about it.

Reply:
Seems fine, in the future we might want to give control to the test, similar to `MockQuery#set_response_data`